carbon-design-system/carbon-for-ibm-dotcom
# fix(masthead): prevent megamenus from overlaying global banner

### Related Ticket(s)

Closes #11775 

### Description

Currently, `<c4d-megamenu>` are slotted into a `<ul>` that is `position: fixed;`.  This is what allows them to extend the full width of the viewport on desktop, but it also means they cannot align themselves relative to other elements using CSS alone. At this stage in the jenga tower, there's not a great CSS-only fix for our issue.

So we turn to our good friend JavaScript for assistance. `<c4d-top-nav-menu>` items now report the position of their bottom edge via a `--c4d-masthead-top-nav-menu-bottom-edge` CSS custom property when toggled open. `<c4d-megamenu>` elements reference this prop to position themselves in the exact right place when they get loaded into the DOM.

### Changelog

**New**

- Makes `<c4d-top-nav-menu>` items expose the position of their bottom edge via a `--c4d-masthead-top-nav-menu-bottom-edge` CSS custom property.

**Changed**

- Updates `<c4d-megamenu>` CSS to reference `--c4d-masthead-top-nav-menu-bottom-edge` value to position themselves correctly when they load into the DOM.

### Testing

1. Visit the [Dotcom Shell > With global banner deploy preview story](https://ibmdotcom-webcomponents.s3.us-east.cloud-object-storage.appdomain.cloud/deploy-previews/11837/index.html?path=/story/components-dotcom-shell--with-global-banner)
2. Open an L0 megamenu item, such as "Products", and verify the megamenu's top edge is aligned to the bottom of the L0.
3. Close the menu, scroll down until the L0 is at the top of the window, then reopen the menu and verify it still aligns properly.
4. Test the same conditions at a mobile viewport width.
